opal-shell/src/main/java/org/obiba/opal/shell/commands/AnalyseCommand.java

```diff
@@ -90,14 +94,20 @@ public int execute() {
         String templateName = analyseOptions.getTemplate();
         log.info("Analysing {} table using {} routines.", tibbleName, String.format("%s::%s", pluginName, templateName));
 
-        RAnalysis analysis = RAnalysis.create(analyseOptions.getName(), analyseOptions.getTemplate())
+        RAnalysis.Builder builder = RAnalysis.create(analyseOptions.getName(), analyseOptions.getTemplate())
           .session(sessionHandler.getSession())
           .symbol(RUtils.getSymbol(tibbleName))
-          .parameters(analyseOptions.getParams())
-          .build();
+          .parameters(analyseOptions.getParams());
 
-        analysisService.save(new OpalAnalysis(datasource.getName(), table.getName(), analysis));
+        if (targetValueTable.isView()) {
```

Comment:
Be aware that the "source" table could be a view as well. Checking for analyseOptions.getVariables().isEmpty() would be more relevant.